### PR TITLE
solution

### PR DESCRIPTION
**Pull request by PyCore23**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,47 @@
...
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self,
+                 distance_from_city_center: float,
+                 clean_power: int,
+                 average_rating: float,
+                 count_of_ratings: int,
+                 ) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py]:
```suggestion
    def __init__(
                 self,
                 distance_from_city_center: float,
                 clean_power: int,
                 average_rating: float,
                 count_of_ratings: int,
                 ) -> None:
```

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,47 @@
...
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def calculate_washing_price(self, car: any) -> float:
```
Comment (Bohdan-Salamakha) [inline app/main.py]: invalid typehint for car

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,47 @@
...
+        return round(car.comfort_class
+                     * (self.clean_power - car.clean_mark)
+                     * (self.average_rating
+                        / self.distance_from_city_center), 1)
+
+    def wash_single_car(self, car: any) -> int:
```
Comment (Bohdan-Salamakha) [inline app/main.py]: invalid typehint for car
+
invalid return typehint

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,47 @@
...
+            car.clean_mark = self.clean_power
+
+    def rate_service(self, points: int) -> None:
+        self.average_rating = round(
+            (self.average_rating * self.count_of_ratings + points)
+            / (self.count_of_ratings + 1), 1
+        )
+        self.count_of_ratings += 1
+
+    def serve_cars(self, cars: list) -> int:
```
Comment (Bohdan-Salamakha) [inline app/main.py]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (pavlejviki) [reply app/main.py]: You haven't fixed the formatting. Please review all your methods

**Review by pavlejviki — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1,8 +1,47 @@
 class Car:
-    # write your code here
-    pass
+
+    def __init__(self,
```
Comment (Polyakiv-Andrey) [inline app/main.py]: After opening the parentheses, write all the arguments on a new line, including the first one, the closing bracket is also written on a new line

File: app/main.py [hunk 6]
```diff
@@ -1,8 +1,47 @@
...
+                 brand: str) -> object:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self,
```
Comment (Polyakiv-Andrey) [inline app/main.py]: After opening the parentheses, write all the arguments on a new line, including the first one, the closing bracket is also written on a new line

File: app/main.py [hunk 7]
```diff
@@ -1,8 +1,47 @@
 class Car:
-    # write your code here
-    pass
+
+    def __init__(self,
+                 comfort_class: int,
+                 clean_mark: int,
+                 brand: str) -> object:
```
Comment (Polyakiv-Andrey) [inline app/main.py]: init return None

File: app/main.py [hunk 8]
```diff
@@ -1,8 +1,47 @@
...
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self,
+                 distance_from_city_center: float,
+                 clean_power: int,
+                 average_rating: float,
+                 count_of_ratings: int,
+                 ) -> object:
```
Comment (Polyakiv-Andrey) [inline app/main.py]: init return None

File: app/main.py [hunk 9]
```diff
@@ -1,8 +1,47 @@
...
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def calculate_washing_price(self, car: object[Car]) -> float:
+        return round(car.comfort_class
+                     * (self.clean_power - car.clean_mark)
+                     * (self.average_rating
+                        / self.distance_from_city_center), 1)
+
+    def wash_single_car(self, car: object[Car]) -> None:
```
Comment (Polyakiv-Andrey) [inline app/main.py]: car  is Car, not object[Car]

**Review by Polyakiv-Andrey — CHANGES_REQUESTED** [review]

**Review by Polyakiv-Andrey — APPROVED** [review]
Good job!

**Pull request closed without merging**